cmd/gnotxport/export.go

```diff
@@ -10,45 +12,99 @@ import (
 
 	"github.com/gnolang/gno/pkgs/amino"
 	"github.com/gnolang/gno/pkgs/bft/rpc/client"
-	"github.com/gnolang/gno/pkgs/command"
+	"github.com/gnolang/gno/pkgs/commands"
 	"github.com/gnolang/gno/pkgs/std"
-
 	// XXX better way?
 	_ "github.com/gnolang/gno/pkgs/sdk/auth"
 	_ "github.com/gnolang/gno/pkgs/sdk/bank"
 	_ "github.com/gnolang/gno/pkgs/sdk/vm"
 )
 
-type txExportOptions struct {
-	Remote      string `flag:"remote" help:"Remote RPC addr:port"`
-	StartHeight int64  `flag:"start" help:"Start height"`
-	TailHeight  int64  `flag:"tail" help:"Start at LAST - N"`
-	EndHeight   int64  `flag:"end" help:"End height (optional)"`
-	OutFile     string `flag:"out" help:"Output file path"`
-	Quiet       bool   `flag:"quiet" help:"Quiet mode"`
-	Follow      bool   `flag:"follow" help:"Keep attached and follow new events"`
+type exportCfg struct {
+	rootCfg *config
+
+	startHeight int64
+	tailHeight  int64
+	endHeight   int64
+	outFile     string
+	quiet       bool
+	follow      bool
 }
 
-var defaultTxExportOptions = txExportOptions{
-	Remote:      "localhost:26657",
-	StartHeight: 1,
-	EndHeight:   0,
-	TailHeight:  0,
-	OutFile:     "txexport.log",
-	Quiet:       false,
-	Follow:      false,
+func newExportCommand(rootCfg *config) *commands.Command {
+	cfg := &exportCfg{
+		rootCfg: rootCfg,
+	}
+
+	return commands.NewCommand(
+		"export",
+		commands.Metadata{
+			Name:       "export",
+			ShortUsage: "export [flags] <file>",
+			ShortHelp:  "Export transactions to file",
+		},
+		cfg,
+	)
 }
 
-func txExportApp(cmd *command.Command, args []string, iopts interface{}) error {
-	opts := iopts.(txExportOptions)
-	c := client.NewHTTP(opts.Remote, "/websocket")
-	status, err := c.Status()
+func (c *exportCfg) RegisterFlags(fs *flag.FlagSet) {
+	fs.Int64Var(
+		&c.startHeight,
+		"start",
+		1,
+		"start height",
+	)
+
+	fs.Int64Var(
+		&c.tailHeight,
+		"tail",
+		0,
+		"start at LAST - N",
+	)
+
+	fs.Int64Var(
+		&c.endHeight,
+		"end",
+		0,
+		"end height (optional)",
+	)
+
+	fs.StringVar(
+		&c.outFile,
+		"out",
+		defaultFilePath,
+		"output file path",
+	)
+
+	fs.BoolVar(
+		&c.quiet,
+		"quiet",
+		false,
+		"omit console output during execution",
+	)
+
+	fs.BoolVar(
+		&c.follow,
+		"follow",
+		false,
+		"keep attached and follow new events",
+	)
```

Comment:
maybe it's just me, but I think this should stay in single lines, for readability.

Reply:
You're right, for short flag declarations like these, we should keep it single-line

Fixed:

[797f8df](https://github.com/gnolang/gno/pull/497/commits/797f8dfa240953d2d326c82d98d8566f176bf824)